internal/backend/remote-state/gcs/backend_internal_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package gcs
+
+import (
+	"encoding/base64"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/internal/backend"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func TestBackendConfig_encryptionKey(t *testing.T) {
+	preCheckEnvironmentVariables(t)
+	// Cannot use t.Parallel as t.SetEnv used
+
+	// getWantValue is required because the key input is changed internally in the backend's code
+	// This function is a quick way to help us get a want value, but ideally in future the test and
+	// the code under test will use a reusable function to avoid logic duplication.
+	getWantValue := func(key string) []byte {
+		var want []byte
+		if key == "" {
+			want = nil
+		}
+		if key != "" {
```

Comment:
This could just be `else` or not even be a conditional if we just return early in the conditional above.